mods/sethome/init.lua

```diff
@@ -1,65 +1,82 @@
+
+sethome = {}
+
 local homes_file = minetest.get_worldpath() .. "/homes"
 local homepos = {}
 
 local function loadhomes()
-    local input = io.open(homes_file, "r")
-    if input then
+	local input = io.open(homes_file, "r")
+	if input then
 		repeat
-            local x = input:read("*n")
-            if x == nil then
-            	break
-            end
-            local y = input:read("*n")
-            local z = input:read("*n")
-            local name = input:read("*l")
-            homepos[name:sub(2)] = {x = x, y = y, z = z}
-        until input:read(0) == nil
-        io.close(input)
-    else
-        homepos = {}
-    end
+			local x = input:read("*n")
+			if x == nil then
+				break
+			end
+			local y = input:read("*n")
+			local z = input:read("*n")
+			local name = input:read("*l")
+			homepos[name:sub(2)] = {x = x, y = y, z = z}
+		until input:read(0) == nil
+		io.close(input)
+	end
 end
 
 loadhomes()
 
-minetest.register_privilege("home", "Can use /sethome and /home")
+sethome.set = function(name, pos)
+	local player = minetest.get_player_by_name(name)
+	if not player or not pos then
+		return false
+	end
+
+	local data = {}
+	local output = io.open(homes_file, "w")
```

Comment:
This needs proper IO handling. When there were problems with opening the file (permission issues for example) it will crash at `:write` later, but hide, similar to the case in line 8, the error message explaining to the user, what went wrong.